src/modules/iotjs_module_mqtt.c

```diff
@@ -74,24 +69,30 @@ static size_t get_remaining_length_size(uint32_t len) {
 }
 
 
-static uint32_t iotjs_decode_remaining_length(char *buffer, size_t *offset) {
-  unsigned char c;
-  uint32_t remaining_length = 0;
+static uint32_t iotjs_decode_remaining_length(char *buffer,
+                                              uint32_t *out_length) {
+  uint32_t decoded_length = 0;
   uint32_t length = 0;
   uint32_t shift = 0;
+  uint32_t c;
 
   do {
     if (++length > IOTJS_MODULE_MQTT_MAX_REMAINING_LENGTH_BYTES) {
-      return UINT32_MAX;
+      return 0;
     }
-    c = (unsigned char)buffer[*offset];
-    remaining_length += (uint32_t)(c & 0x7F) << shift;
+    c = (uint32_t)(*buffer);
+    decoded_length += (c & 0x7F) << shift;
+
     shift += 7;
+    buffer++;
+  } while ((c & 0x80) != 0);
 
-    (*offset)++;
-  } while ((c & 128) != 0);
+  if (c == 0) {
+    return 0;
```

Comment:
Add comment:
// The length must be encoded with the least amount of bytes.
// This rule is not fulfilled if the last byte is zero.